app/src/main/java/com/bilal/cactus_notes_project/ui/SignupActivity.kt

```diff
@@ -0,0 +1,95 @@
+package com.bilal.cactus_notes_project.ui
+
+import android.os.Bundle
+import android.util.Patterns
+import androidx.appcompat.app.AppCompatActivity
+import com.bilal.cactus_notes_project.R
+import com.bilal.cactus_notes_project.databinding.ActivitySignupBinding
+
+
+class SignupActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySignupBinding
+    private val userNameRegex = "^[a-z0-9_]*$".toRegex()
```

Comment:
Can you refactor all `userName` to `username`?

username is the correct spelling (not "user name")

Reply:
Also, since these regexes are only used in the functions below maybe move them into the functions?

Reply:
DONE